src/angular-datatables.directive.ts

```diff
@@ -62,62 +63,60 @@ export class DataTableDirective implements OnDestroy, OnInit {
     }
   }
 
-  private displayTable(): void {
+  private async displayTable() {
     const self = this;
-    this.dtInstance = new Promise((resolve, reject) => {
-      Promise.resolve(this.dtOptions).then(dtOptions => {
-        // Using setTimeout as a "hack" to be "part" of NgZone
-        setTimeout(() => {
-          // Assign DT properties here
-          let options: ADTSettings = {
-            rowCallback: (row, data, index) => {
-              if (dtOptions.columns) {
-                const columns = dtOptions.columns;
-                // Filter columns with pipe declared
-                const colsWithPipe = columns.filter(x => x.ngPipeInstance && !x.ngTemplateRef);
-                // Iterate
-                colsWithPipe.forEach(el => {
-                  const pipe = el.ngPipeInstance;
-                  // find index of column using `data` attr
-                  const i = columns.findIndex(e => e.data == el.data);
-                  // get <td> element which holds data using index
-                  const rowFromCol = row.childNodes.item(i);
-                  // Transform data with Pipe
-                  const rowVal = $(rowFromCol).text();
-                  const rowValAfter = pipe.transform(rowVal);
-                  // Apply transformed string to <td>
-                  $(rowFromCol).text(rowValAfter);
-                });
+    // resolve dtOptions if Promise was provided.
+    if ('then' in this.dtOptions) this.dtOptions = await Promise.resolve(this.dtOptions);
+    // process request
+    this.dtInstance = this.ngZone.runTask(_ => {
+      // Assign DT properties here
+      let options: ADTSettings = {
+        rowCallback: (row, data, index) => {
+          if (this.dtOptions.columns) {
+            const columns = this.dtOptions.columns;
+            // Filter columns with pipe declared
+            const colsWithPipe = columns.filter(x => x.ngPipeInstance && !x.ngTemplateRef);
+            // Iterate
+            colsWithPipe.forEach(el => {
+              const pipe = el.ngPipeInstance;
+              // find index of column using `data` attr
+              const i = columns.findIndex(e => e.data == el.data);
```

Comment:
__chore__: Same here with the equality.

```suggestion
              const i = columns.findIndex(e => e.data === el.data);
```